Source/Spring2022_Capstone/MoveableActor.h

```diff
@@ -0,0 +1,43 @@
+// Created by Spring2022_Capstone team
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameFramework/Actor.h"
+#include "MoveableActor.generated.h"
+
+UCLASS()
+class SPRING2022_CAPSTONE_API AMoveableActor : public AActor
+{
+	GENERATED_BODY()
+
+public:
+	// Sets default values for this actor's properties
+	AMoveableActor();
+
+protected:
+	// Called when the game starts or when spawned
+	virtual void BeginPlay() override;
+	UPROPERTY(VisibleAnywhere, Category = "Components", meta = (AllowPrivateAccess = true))
+	UStaticMeshComponent *Platform;
+ini
+public:
+	// Called every frame
+	virtual void Tick(float DeltaTime) override;
+
+	
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Movement")
+	float RestTimeBetweenMovements;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Movement")
+	float MovementSpeed;
+
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Movement")
+	TArray<FVector> MovementPoints;
```

Comment:
how's this working? do you have to enter the vectors manually?
maybe we could use scenecomponents instead so we can place the movement points on scene and the actor would just follow them? just a suggestion